catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/UserRepository.java

```diff
@@ -121,8 +125,14 @@ protected void postDelete(User entity) {
 
   @Override
   public User setFields(User user, Fields fields) throws IOException {
+    if (fields.contains("teams")) {
+      user.setTeams(getTeams(user));
+    } else if (fields.contains("groups")) {
+      user.setTeams(getGroupTeams(getTeams(user)));
+    } else {
+      user.setTeams(null);
+    }
```

Comment:
will this code block support calls like `http://localhost:8585/api/v1/users/loggedInUser?fields=teams,groups`?

Reply:
`groups` is not part of `allowedFields`, right?

Reply:
I think we need support for both team,groups separately, I tried this locally but with the current implementation if I am calling groups it is sending data in the team's fields and I am not getting actual teams where the user is part!

Reply:
To avoid confusions, I have added a new API `/api/v1/users/loggedInUser/groupTeams` to fetch all the teams of type 'Group'

Reply:
Thanks, @vivekratnavel, but with the new API it will work for only Logged in users in secure mode but in the non-secure mode we are fetching 1st user from the user's list, in that way we don't have any way to fetch groups team for the particular user!